napari/_qt/_qapp_model/qactions/_window.py

```diff
@@ -0,0 +1,84 @@
+"""Actions related to the 'Window' menu that require Qt."""
+
+from typing import List
+
+from app_model.types import Action, ToggleRule
+
+from ...._app_model.constants import CommandId, MenuGroup, MenuId
+from ....utils.translations import trans
+from ...qt_main_window import Window
+
+
+class ToggleDockWidgetAction(Action):
```

Comment:
Have added this (copying format of `ViewerToggleAction` from #4826) because of DRY - though with only 3 items it is possibly unnecessary, though useful future proofing? Open to changes.

Not sure if it is general enough/should belong in its own module.

Checked and callback and toggle both work.

Reply:
> Edit: If we are happy with adding this class, it will need to be tested.

A test similar to `napari/_app_model/_tests/test_viewer_toggler.py` should probably be okay?